AEPIdentity/Sources/CustomIdentity.swift

```diff
@@ -24,19 +24,46 @@ public class CustomIdentity: Identifiable, Codable {
     ///   - type: Type of the identifier
     ///   - identifier: The identifier
     ///   - authenticationState: Authentication state for the identifier
-    init(origin: String?, type: String?, identifier: String?, authenticationState: MobileVisitorAuthenticationState) {
+    public init(origin: String?, type: String?, identifier: String?, authenticationState: MobileVisitorAuthenticationState) {
         self.origin = origin
         self.type = type
         self.identifier = identifier
         self.authenticationState = authenticationState
     }
 
     enum CodingKeys: String, CodingKey {
+        case origin = "id_origin"
+        case type = "id_type"
+        case identifier = "id"
+        case authenticationState = "authentication_state"
+    }
+
+    /// These only exist to support the case where a `CustomIdentity` may have been saved with incorrect keys in a previous SDK version
+    /// https://github.com/adobe/aepsdk-core-ios/issues/433
+    private enum MigrationCodingKeys: String, CodingKey {
         case origin = "id.origin"
         case type = "id.type"
         case identifier = "id"
         case authenticationState = "authentication.state"
     }
+
+    public required init(from decoder: Decoder) throws {
+        do {
+            // Attempt to decode using the correct keys
+            let values = try decoder.container(keyedBy: CodingKeys.self)
+            origin = try? values.decode(String.self, forKey: .origin)
+            type = try? values.decode(String.self, forKey: .type)
+            identifier = try? values.decode(String.self, forKey: .identifier)
+            authenticationState = try values.decode(MobileVisitorAuthenticationState.self, forKey: .authenticationState)
+        } catch _ {
+            // Decoding failed using correct keys, attempt to decode using migration keys
+            let values = try decoder.container(keyedBy: MigrationCodingKeys.self)
```

Comment:
If decoding fails we attempt to use the old "Migration" keys. @emdobrin

Reply:
I think we should either include this in the migration class or expect a new sync call from the developer for the lost ids.  @shalehaha what do you think? this manifests if migrating from v4/v5->3.0.0.